### ISSUE TITLE
How to show validation errors for FieldArray on change of child attribute (v6.0.0-alpha7)

### ISSUE THREAD

**Issue by tacomanator** [erikras/redux-form#931]
I'm trying out v6.0.0-alpha7 and having an issue in which a validation error set on the FieldArray is not showing when checking/unchecking the child field. Validation runs and sets the error, but the FieldArray component is not re-rendered.

Please excuse the syntax (LiveScript):

```livescript
class Form extends Component

  render: ->
    element FieldArray,
      name: "children"
      component: (children) ->
        div null,

           # log statement only called on first render and when adding new children
           # it's not called when checking Field inside, even though that may clear or result in error
           console.log children.error

           choices.map (child) ->
            element Field,
              name: "#{child}.selected",
              component: "input",
              type: "checkbox"

          element Button,
            on-click: ~> children.push {selected: false}
            "Add child"

form-options =

  form: \group

  validate: ({children}) ->

    selected-count = sum map (.selected), children

    children: 
      if selected-count < 1
      then {_error: "Please select a child"}


module.exports = redux-form form-options <| Form
```

The log statement only fires at the beginning or when a new child is pushed into the FieldArray. It's not logging anything when checking/unchecking the checkbox.

Is this the intended behavior? Do I have to manually ```touch``` the FieldArray field when one of the children is changed?

**Comment by erikras** [comment]
This is a bug. There's a `shouldComponentUpdate` on `FieldArray` that is only updating if the length changes, and it needs to be looking at the sync errors, too. Thanks!

Man, that's some crazy-looking code. 🙈 👀 😆

**Comment by tacomanator** [comment]
Ok, thank you :v:

**Comment by halayli** [comment]
http://redux-form.com/6.0.0-alpha.7/examples/fieldArrays/

If you just set the club name and hit submit the error 'At least one member must be entered' doesn't show up. I am guessing it's for the same reason?

**Comment by erikras** [comment]
Published fix as [`v6.0.0-alpha.9`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-alpha.9).
